hasher-matcher-actioner/hmalib/lambdas/actions/action_evaluator.py

```diff
@@ -104,16 +125,27 @@ def lambda_handler(event, context):
 
 
 def get_actions_to_take(
-    match_message: MatchMessage, action_rules: t.List[ActionRule]
+    match_message: MatchMessage,
+    action_rules: t.List[ActionRule],
+    additional_fields_on_content: t.Set[str],
 ) -> t.Dict[ActionLabel, t.List[ActionRule]]:
     """
     Returns action labels for each action rule that applies to a match message.
     """
     action_label_to_action_rules: t.Dict[ActionLabel, t.List[ActionRule]] = dict()
+
+    content_classifications = {
+        SubmittedContentClassification(field) for field in additional_fields_on_content
+    }
+
+    logger.info(
+        "Adding SubmittedContentClassification Labels: %s", content_classifications
+    )
+
     for banked_signal in match_message.matching_banked_signals:
         for action_rule in action_rules:
             if action_rule_applies_to_classifications(
-                action_rule, banked_signal.classifications
+                action_rule, banked_signal.classifications | content_classifications
```

Comment:
nit: the typing was not clear to me and I was confused by the `|`. I prefer we use `.union()` for future readers unless you feel strongly about it

Reply:
Lol. Syntactic sugar isn't so sugary if no one knows what's happening.

Will change